go/vt/vtgate/planbuilder/testdata/union_cases.txt

```diff
@@ -1520,6 +1559,10 @@ Gen4 error: nesting of unions at the right-hand side is not yet supported
     "Inputs": [
       {
         "OperatorType": "Distinct",
+        "Collations": [
+          "1: binary"
+        ],
+        "ResultColumns": 1,
```

Comment:
The looks like the planner knows about collation, then the query should not contain weight_string column